packages/typespec-azure-resource-manager/src/resource.ts

```diff
@@ -92,6 +94,19 @@ export const $armVirtualResource: ArmVirtualResourceDecorator = (
   }
 };
 
+export const $customAzureResource: CustomAzureResourceDecorator = (
+  context: DecoratorContext,
+  entity: Model,
+) => {
+  const { program } = context;
+  if (isTemplateDeclaration(entity)) return;
+  if (!isLegacyTypeSpec(program, entity)) {
```

Comment:
I think we can just report this diagnostic.  Ideally, we would move this check into a linting rule, so we could eventually exclude it from a future 'arm brownfield'  ruleset

Reply:
created the linting rule :)